server/src/main/java/org/apache/cassandra/sidecar/db/schema/SidecarSchema.java

```diff
@@ -49,64 +35,30 @@
 public class SidecarSchema
 {
     private static final Logger LOGGER = LoggerFactory.getLogger(SidecarSchema.class);
-    protected static final DurationSpec INITIALIZATION_LOOP_DELAY = MillisecondBoundConfiguration.parse("1s");
 
-    private final Vertx vertx;
-    private final PeriodicTaskExecutor periodicTaskExecutor;
     private final SchemaKeyspaceConfiguration schemaKeyspaceConfiguration;
     private final SidecarInternalKeyspace sidecarInternalKeyspace;
-    private final CQLSessionProvider cqlSessionProvider;
-    private final SchemaMetrics metrics;
-    private final ClusterLease clusterLease;
-
-    private boolean isInitialized = false;
+    private final AtomicBoolean isInitialized = new AtomicBoolean(false);
 
     public SidecarSchema(Vertx vertx,
-                         PeriodicTaskExecutor periodicTaskExecutor,
                          SidecarConfiguration config,
-                         SidecarInternalKeyspace sidecarInternalKeyspace,
-                         CQLSessionProvider cqlSessionProvider,
-                         SchemaMetrics metrics,
-                         ClusterLease clusterLease)
+                         SidecarInternalKeyspace sidecarInternalKeyspace)
     {
-        this.vertx = vertx;
-        this.periodicTaskExecutor = periodicTaskExecutor;
         this.schemaKeyspaceConfiguration = config.serviceConfiguration().schemaKeyspaceConfiguration();
         this.sidecarInternalKeyspace = sidecarInternalKeyspace;
-        this.cqlSessionProvider = cqlSessionProvider;
-        this.metrics = metrics;
-        this.clusterLease = clusterLease;
         if (this.schemaKeyspaceConfiguration.isEnabled())
         {
-            configureSidecarServerEventListeners();
+            vertx.eventBus().localConsumer(ON_SIDECAR_SCHEMA_INITIALIZED.address(), ignored -> isInitialized.set(true));
```

Comment:
should we use `EventBusUtils.onceLocalConsumer` here?

Reply:
Yep. Good catch